client/javascripts/services.js

```diff
@@ -1,11 +1,178 @@
-app.service("NutritionService", function($http){
+app.service("NutritionService", function($http) {
   return {
+
+    // Makes request to backend with sample API request
     getData: function() {
       // TO DO: 
       // Form validations
 
-      // Makes request to backend with sample API request
       return $http.get('http://localhost:3000/foodData')
+    },
+
+    // Collects only the data necessary for the data visualization
+    getNutrients: function(data) {
+      return data.nutrients;
+    },
+
+    // Collects only macronutrients
+    getMacros: function(data) {
+      return data.slice(3,9).concat(data[0])
     }
   }
 })
+
+  // Display Pie Chart of Macronutrients
+  // Following tutorial from 
+  // http://zeroviscosity.com/d3-js-step-by-step/step-1-a-basic-pie-chart
+  // TO DO: Move logic DOM manipulations to its own directive!
+app.service("D3Service", function() {
```

Comment:
A `service` in angular should be:

> reusable business logic independent of views

See the [Conceptual Overview](https://docs.angularjs.org/guide/concepts)

This D3Service is all about views and interactions. That you'll typically want to do from directives.